Assets/MRTK/Providers/WindowsMixedReality/Shared/InputHandlers/HPMotionControllerInputHandler.cs

```diff
@@ -0,0 +1,246 @@
+
+        private static readonly ProfilerMarker UpdateControllerPerfMarker = new ProfilerMarker("[MRTK] HPMotionControllerInputHander.UpdateController");
+
+        /// <summary>
+        /// Update the controller data from .
+        /// </summary>
+        public virtual void UpdateController(MotionControllerState controllerState)
+        {
+            using (UpdateControllerPerfMarker.Auto())
+            {
+                for (int i = 0; i < Interactions?.Length; i++)
+                {
+                    switch (Interactions[i].AxisType)
+                    {
+                        case AxisType.None:
+                            break;
+                        case AxisType.Digital:
+                            UpdateButtonData(Interactions[i], controllerState);
+                            break;
+                        case AxisType.SingleAxis:
+                            UpdateSingleAxisData(Interactions[i], controllerState);
+                            break;
+                        case AxisType.DualAxis:
+                            UpdateDualAxisData(Interactions[i], controllerState);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateButtonDataPerfMarker = new ProfilerMarker("[MRTK] HPMotionControllerInputHander.UpdateButtonData");
+
+        /// <summary>
+        /// Update an interaction bool data type from a bool input
+        /// </summary>
+        /// <remarks>
+        /// Raises an Input System "Input Down" event when the key is down, and raises an "Input Up" when it is released (e.g. a Button)
+        /// </remarks>
+        internal virtual void UpdateButtonData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateButtonDataPerfMarker.Auto())
+            {
+                // Handedness must be left or right in order to differentiate between buttons for the left and right hand.
+                MotionControllerHandedness controllerHandedness = controllerState.MotionController.Handedness;
+
+                Debug.Assert(controllerHandedness != MotionControllerHandedness.Unknown);
+                Debug.Assert(interactionMapping.AxisType == AxisType.Digital);
+
+                if (interactionMapping.InputType == DeviceInputType.TriggerPress)
+                {
+                    var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                    interactionMapping.BoolData = triggerData.Equals(1);
+                }
+                else if (interactionMapping.InputType == DeviceInputType.GripPress)
+                {
+                    var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                    interactionMapping.BoolData = gripData.Equals(1);
+                }
```

Comment:
Do these need to be their own cases, or could they be combined with the `switch` below? I guess more specifically, does the Input plug-in provide `GetPressedValue` info for `ControllerInput.Grasp` and `ControllerInput.Trigger`, instead of us needing to check that a float value is 1?

Reply:
This ended up pretty related to https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8794#discussion_r515221932